### PR TITLE
Skip rules based endpoints for most services

### PR DESCRIPTION
**Pull request by dagnir**
## Modifications
We are only enabling rules based endpoints for S3, S3Control, and EventBridge for now in order to roll back incorrect hostprefix handling in the Java SDK. S3, S3Control, and EventBridge are unaffected by this issue because their rules are handwritten, and in the case of S3Control which has a host prefix in the model, it is handled by the rule set.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Review by cenedhryn — APPROVED** [review]

**Comment by zoewangg** [comment]
Integ tests passed

File: codegen-lite-maven-plugin/pom.xml [hunk 1]
```diff
@@ -74,10 +74,6 @@
             <artifactId>maven-project</artifactId>
             <groupId>org.apache.maven</groupId>
         </dependency>
-        <dependency>
```
Comment (cenedhryn) [inline codegen-lite-maven-plugin/pom.xml:77]: Why did we need to remove these?

Comment (zoewangg) [reply codegen-lite-maven-plugin/pom.xml:77]: It started to fail because they were "unused declared dependencies"

Comment (cenedhryn) [reply codegen-lite-maven-plugin/pom.xml:77]: gotcha

Comment (zoewangg) [reply codegen-lite-maven-plugin/pom.xml]: Turns out there was a typo in the pom file, fixed it and reverted the change

**Pull request merged**